pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -0,0 +1,341 @@
+    def extend_imm(self):
+        sign = 0
+        self.imm = self.imm << (32 - self.imm_len)
+        sign = self.imm.bin[0:1:1] # sign = imm[31]
+        self.imm = self.imm >> (32 - self.imm_len)
+        #Signed extension
+        if((sign and not(self.format == "U_FORMAT")) or (self.imm_type in ["UIMM", "NZUIMM"])):
+        	self.imm = self.imm_mask | self.imm
+
+
+    def post_randomize(self):
+        self.extend_imm()
+        self.update_imm_str()
+        
+    def convert2asm(self):
+        pass #TODO
+
+    def get_opcode(self):
+    	count =0
+    	if(self.instr_name.name == "LUI"):
+    		count +=1
+    		return (BitArray(uint=55 , length = 7).bin)
+    	if(self.instr_name.name == "AUIPC"):
+    		count +=1
+    		return (BitArray(uint =23 , length = 7).bin)
+    	if(self.instr_name.name == "JAL"):
+    		count +=1
+    		return (BitArray(uint =23 , length = 7).bin)
+    	if(self.instr_name.name == "JALR"):
+    		count +=1
+    		return (BitArray(uint =111 , length = 7).bin)
+    	if(self.instr_name.name in ["BEQ", "BNE", "BLT", "BGE", "BLTU", "BGEU"]):
+    		count +=1
+    		return (BitArray(uint = 103 , length = 7).bin)
+    	if(self.instr_name.name in ["LB", "LH", "LW", "LBU", "LHU", "LWU", "LD"]):
+    		count +=1
+    		return (BitArray(uint = 99 , length = 7).bin)
+    	if(self.instr_name.name in ["SB", "SH", "SW", "SD"]):
+    		count +=1
+    		return (BitArray(uint = 35 , length = 7).bin)
+    	if(self.instr_name.name in ["ADDI", "SLTI", "SLTIU", "XORI", "ORI", "ANDI", "SLLI", "SRLI", "SRAI", "NOP"]):
+    		count +=1
+    		return (BitArray(uint = 19 , length = 7).bin)
+    	if(self.instr_name.name in ["ADD", "SUB", "SLL", "SLT", "SLTU", "XOR", "SRL", "SRA", "OR", "AND", "MUL",\
+    		"MULH", "MULHSU", "MULHU", "DIV", "DIVU", "REM", "REMU"]):
+    		count +=1
+    		return (BitArray(uint = 51 , length = 7).bin)
+    	if(self.instr_name.name in ["ADDIW", "SLLIW", "SRLIW", "SRAIW"]):
+    		count +=1
+    		return (BitArray(uint = 27 , length = 7).bin)
+    	if(self.instr_name.name in ["MULH", "MULHSU", "MULHU", "DIV", "DIVU", "REM", "REMU"]):
+    		count +=1
+    		return (BitArray(uint = 51 , length = 7).bin)
+    	if(self.instr_name.name in ["FENCE", "FENCE_I"]):
+    		count +=1
+    		return (BitArray(uint = 15 , length = 7).bin)
+    	if(self.instr_name.name in ["ECALL", "EBREAK", "CSRRW", "CSRRS", "CSRRC", "CSRRWI", "CSRRSI", "CSRRCI"]):
+    		count +=1
+    		return (BitArray(uint = 115 , length = 7).bin)
+    	if(self.instr_name.name in ["ADDW", "SUBW", "SLLW", "SRLW", "SRAW", "MULW", "DIVW", "DIVUW", "REMW", "REMUW"]):
+    		count +=1
+    		return (BitArray(uint = 59 , length = 7).bin)
+    	if(self.instr_name.name in ["ECALL", "EBREAK", "URET", "SRET", "MRET", "DRET", "WFI", "SFENCE_VMA"]):
+    		count +=1
+    		return (BitArray(uint = 115 , length = 7).bin)
+    	if(count==0):
+    		logging.critical("Unsupported instruction %0s", self.instr_name.name)
+
+
+    def get_func3(self):
+    	count =0
```

Comment:
same comment here, if you make this a series of `if...elif...elif...else` statements you will not need the unnecessary `count` variable

Reply:
Removed the unnecessary count variable.